orcid-core/src/main/java/org/orcid/core/manager/impl/ProfileEntityCacheManagerImpl.java

```diff
@@ -81,11 +82,10 @@ public void put(ProfileEntity profileEntity) {
     }
     public void put(String orcid, ProfileEntity profile) {
         try {
-            synchronized (lockers.obtainLock(orcid)) {
-                profileCache.put(new Element(new OrcidCacheKey(orcid, releaseName), profile));
-            }
+            profileCache.acquireWriteLockOnKey(orcid);
```

Comment:
hum.....if we are replacing `lockers.obtainLock(orcid)) ` then no. If we are saying the original code is incorrect then yes.